lorawan-device/src/region/fixed_channel_plans/join_channels.rs

```diff
@@ -56,19 +58,19 @@ impl JoinChannels {
         self.available_channels = AvailableChannels::default();
     }
 
-    pub(crate) fn get_next_channel(&mut self, rng: &mut impl RngCore) -> usize {
+    pub(crate) fn get_next_channel(&mut self, rng: &mut impl RngCore) -> u8 {
         match (self.preferred_subband, self.num_retries.cmp(&self.max_retries)) {
             (Some(sb), Ordering::Less) => {
                 self.num_retries += 1;
                 // pick a  random number 0-7 on the preferred subband
                 // NB: we don't use 500 kHz channels
-                let channel = (rng.next_u32() as usize % 8) + ((sb as usize - 1) * 8);
+                let channel = (rng.next_u32() as u8 % 8) + ((sb as usize - 1) as u8 * 8);
```

Comment:
`rng.next_u32() as u8` feels quite shady. Can you try with a sample program that prints out bunch of  downcasted values?

Reply:
nice catch! i totally forgot about that
https://doc.rust-lang.org/rust-by-example/types/cast.html